src/common.ts

```diff
@@ -393,7 +393,7 @@ export function validateBindingIdentifier(
   // The BoundNames of LexicalDeclaration and ForDeclaration must not
   // contain 'let'. (CatchParameter is the only lexical binding form
   // without this restriction.)
-  if (kind & (BindingKind.Let | BindingKind.Const) && (t & Token.LetKeyword) === Token.LetKeyword) {
+  if (kind & (BindingKind.Let | BindingKind.Const) && (t & Token.Type) === (Token.LetKeyword & Token.Type)) {
```

Comment:
It was a bug made it very sensitive with the value behind Token.LetKeyword. There are two more same fixes in parser.ts.

Note tsc will replace `(Token.LetKeyword & Token.Type)` with single number, so that should not be an issue.